pkg/storage/unified/store_test.go

```diff
@@ -0,0 +1,79 @@
+package unified_test
+
+import (
+	"net/http"
+	"testing"
+
+	claims "github.com/grafana/authlib/types"
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+	"github.com/grafana/grafana/pkg/infra/tracing"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+	"github.com/grafana/grafana/pkg/storage/unified/sql"
+	"github.com/grafana/grafana/pkg/storage/unified/sql/db/dbimpl"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+func TestIntegrationCreate(t *testing.T) {
+	testUser := &identity.StaticRequester{
+		Type:           claims.TypeUser,
+		Login:          "admin",
+		UserID:         1,
+		UserUID:        "u1",
+		OrgRole:        identity.RoleAdmin,
+		IsGrafanaAdmin: true, // can do anything
+	}
+
+	t.Run("happy path: creates a new resource", func(t *testing.T) {
+		// TODO: Make a Spanner-compatible test
```

Comment:
If you just forget about everything related to SQL and just work through the API to create objects, then that should be totally sufficient.

In an integration test for a storage backend, you'd basically want to watch out for the word "SQL" appearing anywhere in your code and then reconsider the approach. Because the backend could be anything, so the test must not be aware of SQL.

We already do have integration tests, couldn't you just adopt their approach?

Reply:
Absolutely can :) I aimed to just validate the code would work at all in an easy manner. I'll migrate to a proper integration test within that setup.